salt/modules/boto_ec2.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+'''
+Connection module for Amazon EC2
+
+.. versionadded:: TBD
+
+:configuration: This module accepts explicit EC2 credentials but can also
+    utilize IAM roles assigned to the instance trough Instance Profiles.
+    Dynamic credentials are then automatically obtained from AWS API and no
+    further configuration is necessary. More Information available at::
+
+       http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+
+    If IAM roles are not used you need to specify them either in a pillar or
+    in the minion's config file::
+
+        ec2.keyid: GKTADJGHEIQSXMKKRBJ08H
+        ec2.key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+    A region may also be specified in the configuration::
+
+        ec2.region: us-east-1
+
+    If a region is not specified, the default is us-east-1.
+
+    It's also possible to specify key, keyid and region via a profile, either
+    as a passed in dict, or as a string to pull from pillars or minion config:
+
+        myprofile:
+            keyid: GKTADJGHEIQSXMKKRBJ08H
+            key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+            region: us-east-1
+
+:depends: boto
+
+'''
+
+# Import Python libs
+from __future__ import absolute_import
+import logging
+import time
+from distutils.version import LooseVersion as _LooseVersion  # pylint: disable=import-error,no-name-in-module
+
+# Import Salt libs
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+# Import third party libs
+import salt.ext.six as six
+# pylint: disable=import-error
+try:
+    # pylint: disable=import-error
+    import boto
+    import boto.ec2
+    # pylint: enable=import-error
+    logging.getLogger('boto').setLevel(logging.CRITICAL)
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+# pylint: enable=import-error
+
+
+def __virtual__():
+    '''
+    Only load if boto libraries exist and if boto libraries are greater than
+    a given version.
+    '''
+    required_boto_version = '2.8.0'
+    # the boto_ec2 execution module relies on the connect_to_region() method
+    # which was added in boto 2.8.0
+    # https://github.com/boto/boto/commit/33ac26b416fbb48a60602542b4ce15dcc7029f12
+    if not HAS_BOTO:
+        return False
+    elif _LooseVersion(boto.__version__) < _LooseVersion(required_boto_version):
+        return False
+    else:
+        return True
+
+
+def find_instances(instance_id=None, name=None, tags=None, region=None,
+                   key=None, keyid=None, profile=None, **kwargs):
```

Comment:
I'll fix this then, per our offline discussion. I'm going to cancel the merge request until then.